website/docs/language/import/index.mdx

```diff
@@ -0,0 +1,136 @@
+import {
+  to = aws_instance.example
+  id = "i-abcd1234"
+}
+
+resource "aws_instance" "example" {
+  name = "hashi"
+  # (other resource arguments...)
+}
+```
+
+The above `import` block defines an import of the AWS instance with the ID "i-abcd1234" into the `aws_instance.example` resource in the root module.
+
+The `import` block has the following arguments:
+ - `to` - The resource instance address this import is targeting.
+ - `id` - A string with the import ID of the resource.
+ - `provider` (optional) - An optional custom resource provider, see [The Resource provider Meta-Argument](/terraform/language/meta-arguments/resource-provider) for details.
+
+If you do not give the `import` block a `provider` argument, Terraform attempts to import from the default provider by default. 
+
+### Import ID
+
+To locate the resource you want to import, Terraform requires an import ID in the `id` argument. This argument must currently be supplied as a literal string.
+
+The import block's `id` argument is the resource's import ID. The identifier you use for the `id` argument is resource-specific. You can find the required ID in the [provider documentation](https://registry.terraform.io/browse/providers) for the resource you wish to import.
+
+## Plan and Apply an Import
+
+Terraform processes the `import` block during the plan stage. Along with create, update, remove, and move, import is an action Terraform performs on a resource. Once a plan is approved, Terraform imports the resource into its state during the subsequent apply stage.
+
+You can use `import` blocks by doing the following:
+1. Define an `import` block for the resource(s).
+1. Add a `resource` block in your configuration.
+1. Run `terraform plan` to see how Terraform plans to import the resource(s).
+1. Apply the configuration and have Terraform import resources and update your Terraform state.
+
+> **Hands-on:** Try the [State Import](/terraform/tutorials/state/state-import?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS) tutorial.
+
+The `import` block is [_idempotent_](https://en.wikipedia.org/wiki/Idempotence), meaning that applying an import action and running another plan will not generate another import action. 
+
+Terraform only needs to import a resource into a single address once. Attempting to import a resource into the same address again is a (harmless) no-op. For this reason, you can remove `import` blocks after applying or safely leave them in your configuration as a record of the resource's origin for future module maintainers. For more information on maintaining configurations over time, see [Refactoring](/terraform/language/modules/develop/refactoring).
+
+## Resource Configuration
+
+When Terraform imports a resource, it needs you to define that resource in configuration before saving it to state. You can do this by either writing a `resource` block or by [generating configuration](/terraform/language/import/generating-configuration). 
```

Comment:
i want to rephrase this a little so the ordering is clear.

is the following correct? To prevent destroying the resource, you must create a corresponding `resource` block for any `import` block before running an `apply` for the imported resource. You can create resource configuration manually or generate it using Terraform (link)

Reply:
"To prevent destroying the resource" is possibly misleading, because Terraform will not let you get that far. If you add an `import` block without corresponding `resource` config, it will just fail at plan with an error telling you that.

Reply:
Okay, I tried distilling the order Rita suggested with you above point @kmoe. How does this feel?
```suggestion
Before importing, you must add configuration for every resource you want Terraform to import. Otherwise, Terraform throws an error during planning, insisting you add resource configuration before it can successfully import. You can create resource configuration manually or [generate it using Terraform](/terraform/language/import/generating-configuration). 
```